src/Groups/matrices/MatGrp.jl

```diff
@@ -911,19 +919,23 @@ const SU = special_unitary_group
 
 
 """
-    matrix_group(V::T...) where T<:MatrixGroup
-    matrix_group(V::AbstractVector{T}) where T<:MatrixGroup
+    matrix_group(m::Int, R::Ring, V::T...) where T<:Union{MatElem,MatrixGroupElem}
+    matrix_group(m::Int, R::Ring, V::AbstractVector{T}) where T<:Union{MatElem,MatrixGroupElem}
+    matrix_group(V::T...) where T<:Union{MatElem,MatrixGroupElem}
+    matrix_group(V::AbstractVector{T}) where T<:Union{MatElem,MatrixGroupElem}
 
-Return the matrix group generated by elements in the vector `V`.
+Return the matrix group generated by matrices in `V`. If the degree `m` and
+coefficient ring `R` are not given, then `V` must be non-empty
 """
-function matrix_group(V::AbstractVector{T}) where T<:Union{MatElem,MatrixGroupElem}
-   if T<:MatElem      # if T <: MatrixGroupElem, we can already assume that det(V[i]) != 0 
-      for v in V @assert det(v) !=0 "The matrix is not invertible" end
-   end
-   return MatrixGroup(nrows(V[1]),base_ring(V[1]),V)
+function matrix_group(m::Int, R::Ring, V::AbstractVector{T}; check::Bool=true) where T<:Union{MatElem,MatrixGroupElem}
+   return MatrixGroup(m, R, V)
 end
 
-matrix_group(V::T...) where T<:Union{MatElem,MatrixGroupElem} = matrix_group(collect(V))
+matrix_group(m::Int, R::Ring, V::T...; check::Bool=true) where T<:Union{MatElem,MatrixGroupElem} = matrix_group(m, R, collect(V); check)
```

Comment:
Could we change the ordering of the ring and the degree, so `matrix_group(::Ring, ::Int, ::T...)` to align this with `matrix`, `matrix_algebra` and `matrix_space`? (I appreciate that this might be a bigger change, I just wanted to ask for opinions.)

Reply:
On the other hand it would then clash with eg `GL(n,R)` and *that* is a book convention I wouldn't want to break with.

of course we could also support both orders...

Reply:
(Overall I agree with you, though)

Reply:
Perhaps I can discuss this with @fieker tomorrow and we'll resolve it one way or another.

(In theory we could actually offer both orders, the type system would make it possible. Not sure that's a good idea, though....)